scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,160 @@
+    trace_to_testjson: str = None
+    trace_toperfetto: str = None
+
+    def print(self):
+        print(repr(self))
+
+
+class Metadata_Reader:
+    # This class will be initialized with
+    # the name of the script folder and the environment
+    # file name
+    
+    def __init__(self, script_folder, env_file_name):
+        self.script_folder = script_folder
+        self.env = self.__build_env_object__(env_file_name)
+
+
+    # builds the environment object 
+    def __build_env_object__(self, env_file_name):
+        
+        with open(env_file_name) as stream:
+            try:
+                env = yaml.safe_load(stream)
+            except yaml.YAMLError as yaml_exception:
+                raise yaml_exception
+
+        return env
+
+    # resolves the run arguments associated with the environment  
+    def __resolve_env_vals__(self, metadata_dict):
+
+        for run_arg in metadata_dict:
+
+            run_arg_val = metadata_dict[run_arg]
+
+            if not type(run_arg_val)==str or run_arg=="run":
+                metadata_dict[run_arg]=run_arg_val
+                continue
+            
+            if run_arg_val == None:
+                continue
+
+            sub_args = run_arg_val.split('/')
+                        
+            if len(sub_args)==1:
+                run_arg_val=self.env.get(sub_args[0])
+                
+            elif len(sub_args)==2:
+                run_arg_val=self.env.get(sub_args[0]).get(sub_args[1])
+
+            # if a argument has been specified in the comment header
+            # but can't be found in the env file, consider it to be
+            # boolean value.
+            if run_arg_val == None:
+                run_arg_val = True
+
+            if not self.__is_run_arg_valid__(run_arg_val):
+                raise Exception(str(run_arg_val)+" is not a valid value for "+str(run_arg))
+
+            metadata_dict[run_arg] = run_arg_val
+
+
+    # determines if the defined run arguments are valid
+    def __is_run_arg_valid__(self, run_arg_val):
+        return True
+
+
+    # reads the test script file and parses out the run arguments defined in the file
+    def __parse_script__(self, py_script_path, runs_metadata):
+        runs_def_ptrn=re.compile(r'^\s*#\s*test-runner-runs:\s*(.*)$')
+        args_def_ptrn=re.compile(r'^\s*#\s*test-runner-run/([a-zA-Z0-9_]+)/(script|app):\s*(.*)$')
+
+        runs_arg_lines = {}
+        
+    
+        with open(py_script_path, 'r', encoding='utf8') as py_script:
+            for line in py_script.readlines():
+            
+                runs_match = runs_def_ptrn.match(line.strip())
+                args_match = args_def_ptrn.match(line.strip())
+            
+                if runs_match:
+                    for run in runs_match.group(1).strip().split():
+                        runs_arg_lines[run]=[]
+
+                elif args_match:
+                    runs_arg_lines[args_match.group(1)].append(args_match.group(3))
```

Comment:
this loses the 2nd argument i.e. script or app. Why? If we do not need that part, how about not making it part of the syntax?